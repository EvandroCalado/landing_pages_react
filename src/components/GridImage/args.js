export const args = {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.',
  grid: [
    {
      altText: 'Teste 1',
      image:'https://source.unsplash.com/random/800x800?r=1',
    },
    {
      altText: 'Teste 2',
      image:'https://source.unsplash.com/random/800x800?r=2',
    },
    {
      altText: 'Teste 3',
      image:'https://source.unsplash.com/random/800x800?r=3',
    },
    {
      altText: 'Teste 4',
      image:'https://source.unsplash.com/random/800x800?r=4',
    },
    {
      altText: 'Teste 5',
      image:'https://source.unsplash.com/random/800x800?r=5',
    },
    {
      altText: 'Teste 6',
      image:'https://source.unsplash.com/random/800x800?r=6',
    },
  ],
};
