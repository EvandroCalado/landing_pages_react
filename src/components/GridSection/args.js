export const args = {
  background: false,
  title: 'My grid',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.',
  grid: [
    {
      title: 'Teste 1',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
    },
    {
      title: 'Teste 2',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
    },
    {
      title: 'Teste 3',
      description:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip.',
    },
  ],
};
