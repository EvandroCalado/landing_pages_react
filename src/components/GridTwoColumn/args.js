export const args = {
  title: 'Grid Two Column',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur, omnis qui cumque reiciendis accusantium reprehenderit optio voluptates incidunt. Fugit facilis debitis officia repellat ab laudantium voluptatibus incidunt expedita a nihil!`,
  image: "assets/images/javascript.svg"
}