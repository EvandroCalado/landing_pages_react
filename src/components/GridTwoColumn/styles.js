import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 2fr 3fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.xlarge};
      text-align: left;
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.medium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`;

export const ImageContainer = styled.div`
  ${() => css``}
`;

export const Image = styled.img`
  ${() => css``}
`;
