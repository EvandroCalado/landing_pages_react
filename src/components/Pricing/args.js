export const args = {
  id: 1,
  __component: 'section.pricing',
  description_one:
    'The release of Apple Silicon-based Macs at the end of last year generated a flurry of news coverage and some surprises at the machine’s performance. This post details some background information on the experience of porting Firefox to run natively on these CPUs.',
  description_two:
    'We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.',
  title: 'pricing',
  table: [
    {
      id: 1,
      head: 'title 1',
      body: 'content1',
      footer: 'footer 1',
    },
    {
      id: 2,
      head: 'title 2',
      body: 'content2',
      footer: 'footer 2',
    },
    {
      id: 3,
      head: 'title 3',
      body: 'content3',
      footer: 'footer 3',
    },
    {
      id: 4,
      head: 'title 4',
      body: 'content4',
      footer: 'footer 4',
    },
    {
      id: 5,
      head: 'title 5',
      body: 'content5',
      footer: 'footer 5',
    },
  ],
  matadata: {
    id: 7,
    name: 'pricing',
    section_id: '#pricing',
    background: false,
  },
};
