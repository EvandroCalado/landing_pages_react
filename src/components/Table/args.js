export const args = [
  {
    id: 1,
    head: "title 1",
    body: "content 1",
    footer: "footer 1"
  },
  {
    id: 2,
    head: "title 2",
    body: "content 2",
    footer: "footer 2"
  },
  {
    id: 3,
    head: "title 3",
    body: "content 3",
    footer: "footer 3"
  },
  {
    id: 4,
    head: "title 4",
    body: "content 4",
    footer: "footer 4"
  },
  {
    id: 5,
    head: "title 5",
    body: "content 5",
    footer: "footer 5"
  }
]