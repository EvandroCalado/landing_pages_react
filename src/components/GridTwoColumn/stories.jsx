import GridTwoColumn from '.';
import { args } from './args';

const GridTwoColumnStories = {
  title: 'GridTwoColumn',
  component: GridTwoColumn,
  args: args,
  argTypes: {
    children: { type: 'string' },
  },
};

export const Template = (args) => {
  return (
    <div>
      <GridTwoColumn {...args} />
    </div>
  );
};

export default GridTwoColumnStories;
