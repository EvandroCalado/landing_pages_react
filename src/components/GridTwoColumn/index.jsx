import P from 'prop-types';
import * as Styled from './styles';
import SectionBackground from '../SectionBackground';
import Heading from '../Heading';
import TextComponent from '../TextComponent';

const GridTwoColumn = ({ title, text, background = false, image }) => {
  return (
    <SectionBackground background={background}>
      <Styled.Container>
        <Styled.ImageContainer>
          <Heading uppercase colorDark={!background}>
            {title}
          </Heading>
          <TextComponent>{text}</TextComponent>
        </Styled.ImageContainer>
        <Styled.Image src={image} alt={title} />
      </Styled.Container>
    </SectionBackground>
  );
};

GridTwoColumn.propTypes = {
  title: P.string.isRequired,
  text: P.string.isRequired,
  image: P.string.isRequired,
  background: P.bool,
};

export default GridTwoColumn;
