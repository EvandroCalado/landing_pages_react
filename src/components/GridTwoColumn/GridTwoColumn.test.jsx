import GridTwoColumn from '.';
import { renderTheme } from '../../styles/render.theme';

import { args } from './args';

describe('<GridTwoColumn />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumn {...args} />);
    expect(container).toMatchSnapshot();
  });
});
