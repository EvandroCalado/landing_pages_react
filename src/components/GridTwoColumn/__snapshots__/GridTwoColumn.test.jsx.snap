// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render two column grid 1`] = `
.c4 {
  color: #0a1128;
  font-size: 6.4rem;
  font-weight: 900;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c3 {
  margin-bottom: 4rem;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #fff;
  color: #0a1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c4 {
    font-size: 4rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class=""
        >
          <h1
            class="c3 c4"
          >
            Grid Two Column
          </h1>
          <p
            class="c5"
          >
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Consequatur, omnis qui cumque reiciendis accusantium reprehenderit optio voluptates incidunt. Fugit facilis debitis officia repellat ab laudantium voluptatibus incidunt expedita a nihil!
          </p>
        </div>
        <img
          alt="Grid Two Column"
          class=""
          src="assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
